### ISSUE TITLE
[FEATURE REQUEST] glib: Add functions to get glib version

### ISSUE THREAD

**Issue by A6GibKm** [gtk-rs/gtk-rs-core#1545]
Provide equivalents of:
- GLIB_MAJOR_VERSION
- GLIB_MINOR_VERSION
- GLIB_MICRO_VERSION

This could be helpful, for example, to print a debug message if glib_check_version fails.

**Comment by sdroege** [comment]
This needs to be added to `glib/sys/src/manual.rs` then. Special care is needed to avoid re-opening https://github.com/gtk-rs/gtk-rs-core/pull/1508

@eerii can you confirm (and write down how to check) if these variables are OK and don't suffer from the same problem?

**Comment by sdroege** [comment]
```cpp
GLIB_VAR const guint glib_major_version;
GLIB_VAR const guint glib_minor_version;
GLIB_VAR const guint glib_micro_version;
GLIB_VAR const guint glib_interface_age;
GLIB_VAR const guint glib_binary_age;
```

from `glib/gversion.h` FWIW

**Comment by sdroege** [comment]
For comparison, the bad one is

```cpp
GOBJECT_VAR GType *g_param_spec_types;
```

**Comment by eerii** [comment]
I think these should be fine since they are constants, `g_param_spec_types` being a static pointer was the problematic bit.

The version constants are already in [glib/sys/Gir.toml](https://github.com/gtk-rs/gtk-rs-core/blob/master/glib/sys/Gir.toml#L39), but they are marked as ignored. I tried removing them from this list and regenerating the bindings, and it seems to generate fine c:

```rs
// glib/sys/src/lib.rs
pub const GLIB_MAJOR_VERSION: c_int = 2;
pub const GLIB_MICRO_VERSION: c_int = 0;
pub const GLIB_MINOR_VERSION: c_int = 83;
```

**Comment by eerii** [comment]
For future reference, the issue comes from Windows not including the unprefixed version of exported **variables** (they do so for functions), and Rust only being able to pick up the prefixed version if the `#[link]` attribute with an explicit library name is added to the extern block of the bindings.

To check this, you can compile or download the glib `dll` from [the archive](https://download.gnome.org/binaries/win32/glib/) (in this case, it is in `lib/libglib-2.0.dll.a`), and use `dumpbin.exe /EXPORTS <dll>` on Windows or `winedump -j export <dll>` on Linux to see the symbol names that are exported.

**Comment by sdroege** [comment]
Ok, so we can't provide access to these variables then because GLib is exporting them wrongly.

**Comment by eerii** [comment]
I don't think it really an issue with GLib exporting them wrongly. It is more a combination of the Windows compiler deciding that variables should be explicitly imported with `__dllimport` but not requiring it for functions, and that Rust currently has the limitation of only generating these symbols for blocks annotated with `#[link]`, while not providing as much flexibility with it as with command line linking. So excluding using workarounds or proposing alternatives in Rust there's not much more we can do at the moment.
